src/test_utils.jl

```diff
@@ -31,17 +31,18 @@ mutating the field.
 - [`fit`](@ref) does not learn anything, it just creates an instance of the corresponding [`Model`](@ref).
 - [`predict`](@ref) applies the `predictor` to the inputs.
 """
-mutable struct FakeTemplate{E<:EstimateTrait, O<:OutputTrait} <: Template
+mutable struct FakeTemplate{E<:EstimateTrait, O<:OutputTrait, I<:PredictInputTrait} <: Template
```

Comment:
why add this as a type parameter?

one of the benefits of traits is that adding new traits _doesn't_ require changing the struct to include this info. 

I think it'd be cleaner if this PR didn't alter `FakeTemplate` at all

(also i suppose technically this could be breaking, since `FakeTemplate{SomeEstimateType, SomeOutputType}` is no longer a concrete type, but i doubt that matters)

Reply:
I thought I had to add it because of these definitions:
```
Models.estimate_type(::Type{<:FakeModel{E, O, I}}) where {E, O, I} = E
Models.output_type(::Type{<:FakeModel{E, O, I}}) where {E, O, I} = O
Models.predict_input_type(::Type{<:FakeModel{E, O, I}}) where {E, O, I} = I
```
If the `PredictInputTrait` wasn't a type parameter wouldn't I have to define a new subtype of `FakeModel` for each subtype of the `PredictInputTrait`?